sdkv2/src/main/kotlin/com/nohrd/bike/sdkv2/ByteStuffer.kt

```diff
@@ -0,0 +1,63 @@
+package com.nohrd.bike.sdkv2
+
+/**
+ * This object can be used to decode/encode messages that are sent or received from the v2 Nohrd Bike through USB
+ */
+public object ByteStuffer {
+
+    public fun stuff(input: ByteArray): ByteArray {
+        val stuffed = input
+            .flatMap {
+                when (it) {
+                    0xF1.toByte() -> listOf(0xF3.toByte(), 0x01.toByte())
+                    0xF2.toByte() -> listOf(0xF3.toByte(), 0x02.toByte())
+                    0xF3.toByte() -> listOf(0xF3.toByte(), 0x03.toByte())
+                    else -> listOf(it)
+                }
+            }
+            .toByteArray()
+
+        return byteArrayOf(0xF1.toByte()) + stuffed + 0xF2.toByte()
+    }
+
+    public fun unstuff(input: ByteArray): ByteArray {
+        assert(input.firstOrNull() == 0xF1.toByte())
+        assert(input.lastOrNull() == 0xF2.toByte())
+
+        return input
+            .drop(1)
```

Comment:
Stuffing and unstuffing should only be applied to the "body" of the packet, so the unstuffer shouldn't be the one that drops the start and end packets

Reply:
Renamed the object to UsbCodec